### PR TITLE
MULE-16668: Excessive validations on pooled connections.

### PR DESCRIPTION
**Pull request by ndinu**

### CODE REVIEW

File: core/src/main/java/org/mule/runtime/core/internal/connection/PoolingConnectionHandler.java [hunk 1]
```diff
@@ -68,10 +68,6 @@ public void release() {
 
     boolean returnAttempted = false;
     try {
-      if (!connectionProvider.validate(connection).isValid()) {
-        LOGGER.debug("Connection is not valid, it is destroyed and not returned to the pool.");
-        return;
-      }
       poolingListener.onReturn(connection);
 
       pool.returnObject(connection);
```
Comment (marianogonzalez) [inline core/src/main/java/org/mule/runtime/core/internal/connection/PoolingConnectionHandler.java:73]: this needs to happen on a finally clause. If line 71 files, this needs to be executed anyway

**Comment by ndinu** [comment]
--validate

**Comment by ndinu** [comment]
--validate

**Comment by ndinu** [comment]
--validate

**Pull request merged**